backend/src/main/java/ai/verta/modeldb/cron_jobs/ParentTimestampUpdateCron.java

```diff
@@ -0,0 +1,113 @@
+package ai.verta.modeldb.cron_jobs;
+
+import ai.verta.modeldb.utils.ModelDBHibernateUtil;
+import java.util.TimerTask;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.query.Query;
+
+public class ParentTimestampUpdateCron extends TimerTask {
+  private static final Logger LOGGER = LogManager.getLogger(ParentTimestampUpdateCron.class);
+
+  public ParentTimestampUpdateCron() {}
+
+  /** The action to be performed by this timer task. */
+  @Override
+  public void run() {
+    LOGGER.info("ParentTimestampUpdateCron wakeup");
+
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      session.beginTransaction();
```

Comment:
Don't create long transactions. Also, none of these SQL commands even require a transaction.

Reply:
These SQL commands are the update operation so for the hibernate we can not execute update operation without the transaction. but ye definitely I can reduce transaction scope and create a new one for each.

Reply:
I'm not sure that is correct. I believe hibernate works without transactions. In any case, as long as we put the `LIMIT`, doing a transaction per set of elements seems like a fair trade.

Reply:
Added `LIMIT` in the query.